### PR TITLE
Workshop: Goblin Breakdown

### PR DESCRIPTION
**Pull request by SallyMcGrath**
I had this idea for Chat GPT a while ago with a prompt but this might be better.

This is to support JS3 week 2, which needs a lot of time in class for TV Show project. I looked for the debugging workshops but there's no explanation or workshoppy bit, just some exercises.  So this workshop is trying to be THIRTY MINUTES. (omg she's a dreamer)

The formatting and breaking down functions obviously is about TV show. I don't want them to get too distracted so have provided full solutions, as simple as pos. I probably would break down the render function further tbh but hey ho.

Blocking JS3-2 dayplan so would appreciate a swift review.

### CODE REVIEW

File: goblin-breakdown/readme.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+### 🧠 4. How can you use this?
+
+Set a whole [class timer for 10 minutes](https://www.google.com/search?q=15+minute+timer).
+
+<!--{{<note type="activity" title="Reflect and Discuss 10m">}}>-->
+
+Reflect on this process of breaking down the problem and planning your solution. Discuss with your pair or group.
+
+- How can you use this process in your own work?
+- What are the benefits of breaking down a problem?
```
Comment (illicitonion) [inline goblin-breakdown/readme.md:70]:
```suggestion
- What are the benefits of breaking down a problem?
- What are the risks of having an AI break down a problem? How can you mitigate those risks?
```

File: goblin-breakdown/readme.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+- What are the benefits of planning your solution?
+- Why does CYF want to make you write such small functions?
```
Comment (illicitonion) [inline goblin-breakdown/readme.md:72]:
```suggestion
- Why does CYF want to make you write such small functions?
- What would be easier about implementing the solution to the problem, having done all of these steps, than if you hadn't?
```

File: goblin-breakdown/readme.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+<!--{{</note>}}-->
+
+### 🧠 4. How can you use this?
+
+Set a whole [class timer for 10 minutes](https://www.google.com/search?q=15+minute+timer).
+
+<!--{{<note type="activity" title="Reflect and Discuss 10m">}}>-->
+
+Reflect on this process of breaking down the problem and planning your solution. Discuss with your pair or group.
```
Comment (illicitonion) [inline goblin-breakdown/readme.md:67]: One more reflection I'm not sure how to prompt - I'm generally of the view that this kind of breaking down _gives you easier sub-problems to Google / ask ChatGPT_.

"How do I set the text of a created HTML element?" - easy to Google
"How do I show an address in a page?" - harder to Google

I'm not sure there's an easy way to prompt this reflection without having the trainees actually do the implementation, but I'd be curious as to your thoughts...

File: goblin-breakdown/readme.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+## 👣 2. Break down coding problems
+
+Let's try breaking down a coding problem. Let's do a sample problem together. Set a whole [class timer for 10 minutes](https://www.google.com/search?q=15+minute+timer).
+
+<!--{{<note type="activity" title="Planning a Solution 10m">}}>-->
+
+### 🧱 3. Plan a solution
+
+> Format a UK address and UK phone number from a JSON object and show it as HTML on the page
```
Comment (illicitonion) [inline goblin-breakdown/readme.md:46]: I think we should explicitly call out the schema (which can be as simple as saying "There is an example of the JSON returned from the API in `data.json`

I think "formatting an address" when you don't know whether the address comes as one string, or a string per line, or as even more structured fields, is unspecified for this particular workshop in a way which probably hinders more than it helps :)

File: goblin-breakdown/readme.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+1. Paste the above problem into Goblin and press the 🪄 wand button.
+1. Goblin will give you some steps. In the `contact` folder, paste each of the steps into `index.js` as comments. There are some samples given but your Goblin might be a bit different.
```
Comment (illicitonion) [inline goblin-breakdown/readme.md:49]:
```suggestion
1. Goblin will give you some steps. In the `contact` folder, paste each of the steps into `index.js` as comments. There are some samples given but your Goblin might be a bit different. Feel free to delete the existing examples.
```

File: goblin-breakdown/readme.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+1.  Set a whole class [timer for 5 minutes](https://www.google.com/search?q=timer+for+5+minutes).
+1.  Fork this repository to your own GitHub account if you have not already.
+1.  Clone the repository to your machine.
+1.  Open in VS Code and go to the `goblin-breakdown` folder.
+
+<!--{{</note>}}-->
+
+## 👣 2. Break down coding problems
+
+Let's try breaking down a coding problem. Let's do a sample problem together. Set a whole [class timer for 10 minutes](https://www.google.com/search?q=15+minute+timer).
```
Comment (SallyMcGrath) [inline goblin-breakdown/readme.md:40]:
```suggestion
Let's try breaking down a coding problem. Let's do a sample problem together. Set a whole [class timer for 10 minutes](https://www.google.com/search?q=10+minute+timer).
```

File: goblin-breakdown/readme.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+<!--{{<note type="activity" title="Reflect and Discuss 10m">}}>-->
+
+Reflect on this process of breaking down the problem and planning your solution. Discuss with your pair or group.
+
+- How can you use this process in your own work?
+- What are the benefits of breaking down a problem?
+- What are the risks of having an AI break down a problem? How can you mitigate those risks?
+- What are the benefits of planning your solution?
+- Why does CYF want to make you write such small functions?
+- What would be easier about implementing the solution to the problem, having done all of these steps, than if you hadn't?
```
Comment (SallyMcGrath) [inline goblin-breakdown/readme.md:74]:
```suggestion
- How could doing these steps make writing your code easier?
```

Grade 11 when I wrote this at Grade 3! Break up your ~functions~ sentences! 😛

File: goblin-breakdown/readme.md (hunk 3 above)
Comment (SallyMcGrath) [reply goblin-breakdown/readme.md:67]: Yeah, I agree it's much easier. We can pick that up in another workshop.

File: goblin-breakdown/readme.md (hunk 4 above)
Comment (SallyMcGrath) [reply goblin-breakdown/readme.md:46]: So, interestingly, if you prompt the tool with the json, the steps given are much more vague. 

format the address and phone number and show them as html on a page { "name": "Mahan Esfahani", "address": "789 Pine St, Sometown, Somewhere", "phone": "+44 7700 999123" }

Gives you more like "Format the address and phone number in the desired output format"

If you use the vaguer framing, it often suggests formatting, like +44 to (0)

**Pull request merged**